bindings/java/src/test/java/org/openucx/jucx/UcpListenerTest.java

```diff
@@ -159,4 +159,73 @@ public void testConnectionHandler() throws Exception {
             serverWorker2, listener);
         closeResources();
     }
+
+    @Test
+    public void testAcceptHandler() throws Exception {
+        UcpContext context1 = new UcpContext(new UcpParams().requestStreamFeature()
+            .requestRmaFeature());
+        UcpContext context2 = new UcpContext(new UcpParams().requestStreamFeature()
+            .requestRmaFeature());
+        UcpWorker serverWorker = context1.newWorker(new UcpWorkerParams());
+        UcpWorker clientWorker = context2.newWorker(new UcpWorkerParams());
+
+        AtomicReference<UcpEndpoint> endpointReference = new AtomicReference<>(null);
+
+        // Create listener and set accept handler
+        UcpListenerParams listenerParams = new UcpListenerParams()
+            .setAcceptHandler(endpointReference::set);
+        UcpListener listener = tryBindListener(serverWorker, listenerParams);
+
+        UcpEndpoint clientToServer = clientWorker.newEndpoint(new UcpEndpointParams()
+            .setSocketAddress(listener.getAddress()));
+
+        while (endpointReference.get() == null) {
+            serverWorker.progress();
+            clientWorker.progress();
+        }
+
+        UcpEndpoint serverToClient = endpointReference.get();
+
+        // Temporary workaround until new connection establishment protocol in UCX.
+        for (int i = 0; i < 10; i++) {
+            serverWorker.progress();
+            clientWorker.progress();
+            try {
+                Thread.sleep(10);
+            } catch (Exception ignored) { }
+        }
+
+        UcpRequest sent = serverToClient.sendStreamNonBlocking(
+            ByteBuffer.allocateDirect(UcpMemoryTest.MEM_SIZE), null);
+
+        // Progress all workers to make sure recv request will complete immediately
+        for (int i = 0; i < 10; i++) {
+            serverWorker.progress();
+            clientWorker.progress();
+            try {
+                Thread.sleep(2);
+            } catch (Exception ignored) { }
+        }
+
+        UcpRequest recv = clientToServer.recvStreamNonBlocking(
+            ByteBuffer.allocateDirect(UcpMemoryTest.MEM_SIZE), 0, null);
+
+        while (!sent.isCompleted() || !recv.isCompleted()) {
+            serverWorker.progress();
+            clientWorker.progress();
+        }
+
+        assertEquals(UcpMemoryTest.MEM_SIZE, recv.getRecvSize());
+
+        UcpRequest serverClose = serverToClient.closeNonBlockingForce();
```

Comment:
It stucks if using `closeFlush` on created endpoint from a listener accept handler.  FYI @evgeny-leksikov  @yosefe

Reply:
1) why need to add deprecated API to JUCX?
2) do you see stuck in non-blocking `ucp_ep_close_nb[x]` call or waiting completion of requests below?

Reply:
agree on point (1) - need to expose only connect-request based api

Reply:
It waits on completion of ecp_ep_close flush request